flycheck.el

```diff
@@ -1149,6 +1147,20 @@ Return the path of the file."
         tempfile)
     (flycheck-temp-file-system filename)))
 
+(defun flycheck-temp-files-writable-p (checker)
+  "Whether CHECKER can write temporary files.
+
+If CHECKER has `source' or `source-inplace' in its `:command',
+return whether flycheck has the permissions to create the
+respective temporary files.
+
+Return t if CHECKER does not use temporary files."
+  (let ((args (flycheck-checker-arguments checker)))
+    (and (or (not (memq 'source args))
+             (file-writable-p (flycheck-temp-file-system buffer-file-name)))
+         (or (not (memq 'source-inplace args))
+             (file-writable-p (flycheck-temp-file-inplace buffer-file-name))))))
+
```

Comment:
One issue with this implementation is that it adds the files to `flycheck-temporaries`.

Reply:
Ah right.  We could `pop flycheck-temporaries`... but that would be bad.

So I guess we should split the existing functions to have one return the temporary filename that would be written, and the other actually create the file.

However, `make-temp-file` actually *creates* the temporary file, so it's still not quite correct.  We might have to wait for an error when creating the files after all.

Reply:
> So I guess we should split the existing functions to have one return the temporary filename that would be written, and the other actually create the file.

I'm always a bit scared of that, because race conditions.  It probably doesn't matter too much for Flycheck, though.

> However, make-temp-file actually creates the temporary file, so it's still not quite correct. We might have to wait for an error when creating the files after all.

Hm. What about just moving the push into flycheck-temporaries to flycheck-save-buffer-to-temp?

Reply:
> Hm. What about just moving the push into flycheck-temporaries to flycheck-save-buffer-to-temp?

That's good I think.  But isn't it better to keep track of these temporaries actually?  At least, if we can clean them at the end of `temporaries-writable-p` that way?

Reply:
Ah, sorry, I didn't understand the make-temp-file comment at first.
… thinks for a while …
Could we just use file-writable-p on the appropriate directory?  A quick look at the C sources suggests that it should work on GNU/Linux systems.

Reply:
> Could we just use file-writable-p on the appropriate directory? 

I'll try that, that seems better in the end yes.